code/game/machinery/PDApainter.dm

```diff
@@ -1,66 +1,106 @@
 /obj/machinery/pdapainter
-	name = "\improper PDA painter"
-	desc = "A PDA painting machine. To use, simply insert your PDA and choose the desired preset paint scheme."
+	name = "\improper color manipulator"
+	desc = "A machine able to color PDAs and IDs with ease. Insert an ID card or PDA and pick a color scheme."
 	icon = 'icons/obj/pda.dmi'
-	icon_state = "pdapainter"
+	icon_state = "coloriser"
 	density = TRUE
 	anchored = TRUE
 	var/obj/item/device/pda/storedpda = null
-	var/list/colorlist = list()
+	var/obj/item/card/id/storedid = null
+	var/pda_icons = list(

```

Comment:
While the ID list is fine, the PDA list can be automated fairly sensibly. 
Put this in the init proc.
```
for(var/path in subtypesof(obj/item/device/pda)
    var/obj/item/device/pda/P = new path
    if(!P.hidden)
        pda_icons[P.name] = P.icon_state
    qdel(P)
```

Reply:
Currently it goes by jobs, and I would rather it be standard to job than have PDA tacked onto it, as that was the reason to change it in the first place